samples/net/sockets/civetweb/src/libc_extensions.c

```diff
@@ -0,0 +1,262 @@
+	STRERR_DEFINE(EIO),
+	STRERR_DEFINE(ENXIO),
+	STRERR_DEFINE(ENOTBLK),
+	STRERR_DEFINE(ENODEV),
+	STRERR_DEFINE(ENOTDIR),
+	STRERR_DEFINE(EISDIR),
+	STRERR_DEFINE(ETXTBSY),
+	STRERR_DEFINE(ENOEXEC),
+	STRERR_DEFINE(EINVAL),
+	STRERR_DEFINE(E2BIG),
+	STRERR_DEFINE(ELOOP),
+	STRERR_DEFINE(ENAMETOOLONG),
+	STRERR_DEFINE(ENFILE),
+	STRERR_DEFINE(EMFILE),
+	STRERR_DEFINE(EBADF),
+	STRERR_DEFINE(ECHILD),
+	STRERR_DEFINE(EFAULT),
+	STRERR_DEFINE(EFBIG),
+	STRERR_DEFINE(EMLINK),
+	STRERR_DEFINE(ENOLCK),
+	STRERR_DEFINE(EDEADLK),
+	STRERR_DEFINE(ECANCELED),
+	STRERR_DEFINE(ENOSYS),
+	STRERR_DEFINE(ENOMSG),
+	STRERR_DEFINE(ENOSTR),
+	STRERR_DEFINE(ENODATA),
+	STRERR_DEFINE(ETIME),
+	STRERR_DEFINE(ENOSR),
+	STRERR_DEFINE(EPROTO),
+	STRERR_DEFINE(EBADMSG),
+	STRERR_DEFINE(ENOTSOCK),
+	STRERR_DEFINE(EDESTADDRREQ),
+	STRERR_DEFINE(EMSGSIZE),
+	STRERR_DEFINE(EPROTOTYPE),
+	STRERR_DEFINE(ENOPROTOOPT),
+	STRERR_DEFINE(EPROTONOSUPPORT),
+	STRERR_DEFINE(ESOCKTNOSUPPORT),
+	STRERR_DEFINE(ENOTSUP),
+	STRERR_DEFINE(EPFNOSUPPORT),
+	STRERR_DEFINE(EAFNOSUPPORT),
+	STRERR_DEFINE(EADDRNOTAVAIL),
+	STRERR_DEFINE(ENETDOWN),
+	STRERR_DEFINE(ENETUNREACH),
+	STRERR_DEFINE(ENETRESET),
+	STRERR_DEFINE(ECONNABORTED),
+	STRERR_DEFINE(ENOBUFS),
+	STRERR_DEFINE(EISCONN),
+	STRERR_DEFINE(ENOTCONN),
+	STRERR_DEFINE(ESHUTDOWN),
+	STRERR_DEFINE(EALREADY),
+	STRERR_DEFINE(EINPROGRESS),
+};
+
+static char *strerr_unknown = "UNKNOWN";
+
+char *strerror(int err)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(error_strings); ++i) {
+		if (error_strings[i].err == err) {
+			return (char *)error_strings[i].errstr;
+		}
+	}
+
+	return strerr_unknown;
+}
+
+int sscanf(const char *s, const char *format, ...)
+{
+	FN_MISSING();
+
+	return 0;
+}
+
+double atof(const char *str)
+{
+	/* XXX good enough for civetweb uses */
+	return (double)atoi(str);
+}
+
+long long int strtoll(const char *str, char **endptr, int base)
+{
+	/* XXX good enough for civetweb uses */
+	return (long long int)strtol(str, endptr, base);
+}
+
+time_t time(time_t *t)
+{
+	return 0;
+}
+
+/*
+ * Most of the wrappers below are copies of the wrappers in net/sockets.h,
+ * but they are available only if CONFIG_NET_SOCKETS_POSIX_NAMES is enabled
+ * which is impossible here.
+ */
+
+int getsockname(int sock, struct sockaddr *addr,
+		socklen_t *addrlen)
+{
+	return zsock_getsockname(sock, addr, addrlen);
+}
+
+int poll(struct zsock_pollfd *fds, int nfds, int timeout)
+{
+	return zsock_poll(fds, nfds, timeout);
+}
+
+int getnameinfo(const struct sockaddr *addr, socklen_t addrlen,
+		char *host, socklen_t hostlen,
+		char *serv, socklen_t servlen, int flags)
+{
+	return zsock_getnameinfo(addr, addrlen, host, hostlen,
+				 serv, servlen, flags);
+}
+
+ssize_t send(int sock, const void *buf, size_t len, int flags)
+{
+	return zsock_send(sock, buf, len, flags);
+}
+
+ssize_t recv(int sock, void *buf, size_t max_len, int flags)
+{
+	return zsock_recv(sock, buf, max_len, flags);
+}
+
+int socket(int family, int type, int proto)
```

Comment:
Why re-declaring all these functions since include/net/socket.h does it already when CONFIG_NET_SOCKETS_OFFLOAD is not set. (and it is, the offloading part provides these anyway).

Reply:
It only provides `zsock_socket`. `socket` and the like are only provided if `CONFIG_NET_SOCKETS_POSIX_NAMES` is enabled which cannot be used together with `POSIX_APIS` in general. So until #16621 is merged we were told to add this stuff to the sample (see #16683 and other issues referenced there).

Reply:
Ok